### PR TITLE
Update suggestion periodically (--suggest-interval-ms) in clickhouse-client/local

### PR DESCRIPTION
**Pull request by azat**
Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Update suggestion periodically (--suggest-interval-ms) in clickhouse-client/local (to get new tables/databases)

### CODE REVIEW

**Comment by azat** [comment]
>Stateless tests (thread, actions) — fail: 1, passed: 3601, skipped: 27

- `01676_long_clickhouse_client_autocomplete` - `send: spawn id exp3 not open` it is racy

**Comment by azat** [comment]
>[Integration tests (thread, actions) [1/4] — fail: 30, passed: 353, flaky: 0](https://s3.amazonaws.com/clickhouse-test-reports/32370/f716a371fb4125384ffe853011586327ec0a2191/integration_tests__thread__actions__[1/4].html)

```
E               Exception: Command ['docker-compose', '--env-file', '/ClickHouse/tests/integration/test_grpc_protocol_ssl/_instances_2/.env', '--project-name', 'roottestgrpcprotocolssl', '--file', '/compose/docker_compose_net.yml', '--file', '/ClickHouse/tests/integration/test_grpc_protocol_ssl/_instances_2/node/docker-compose.yml', 'up', '-d', '--no-recreate'] return non-zero code 1: Building with native build. Learn about native build in Compose here: https://docs.docker.com/go/compose-native-build/
E               Creating network "roottestgrpcprotocolssl_default" with driver "bridge"
E               Pool overlaps with other one on this address space
```

```
E               Exception: Command ['docker-compose', '--env-file', '/ClickHouse/tests/integration/test_version_update/_instances_2/.env', '--project-name', 'roottestversionupdate', '--file', '/ClickHouse/tests/integration/test_version_update/_instances_2/node1/docker-compose.yml', '--file', '/ClickHouse/tests/integration/test_version_update/_instances_2/node2/docker-compose.yml', '--file', '/compose/docker_compose_keeper.yml', '--file', '/ClickHouse/tests/integration/test_version_update/_instances_2/node3/docker-compose.yml', '--file', '/ClickHouse/tests/integration/test_version_update/_instances_2/node4/docker-compose.yml', '--file', '/ClickHouse/tests/integration/test_version_update/_instances_2/node5/docker-compose.yml', '--file', '/ClickHouse/tests/integration/test_version_update/_instances_2/node6/docker-compose.yml', 'up', '-d', '--no-recreate'] return non-zero code 1: Building with native build. Learn about native build in Compose here: https://docs.docker.com/go/compose-native-build/
E               Pulling node5 (yandex/clickhouse-server:21.5)...
E               Get "https://registry-1.docker.io/v2/": net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
```

**Comment by azat** [comment]
Can someone take a look please?

File: src/Client/ClientBase.cpp [hunk 1]
```diff
@@ -1349,15 +1349,14 @@ void ClientBase::runInteractive()
...
 
-    std::optional<Suggest> suggest;
-    suggest.emplace();
+    Suggest suggest(suggest_interval_ms);
     if (load_suggestions)
     {
         /// Load suggestion data from the server.
         if (global_context->getApplicationType() == Context::ApplicationType::CLIENT)
-            suggest->load<Connection>(global_context, connection_parameters, config().getInt("suggestion_limit"));
+            suggest.schedule<Connection>(global_context, connection_parameters, config().getInt("suggestion_limit"));
```
Comment (kssenii) [inline src/Client/ClientBase.cpp:1357]: btw for what type of suggestions this is useful to update periodically?

Comment (azat) [reply src/Client/ClientBase.cpp:1357]: Table and databases names.

File: base/base/LineReader.cpp [hunk 2]
```diff
@@ -36,10 +36,9 @@ bool hasInputData()
 
 }
 
-std::optional<LineReader::Suggest::WordsRange> LineReader::Suggest::getCompletions(const String & prefix, size_t prefix_length) const
+LineReader::Suggest::Words LineReader::Suggest::getCompletions(const String & prefix, size_t prefix_length)
 {
-    if (!ready)
-        return std::nullopt;
+    std::lock_guard lock(mutex);
```
Comment (kssenii) [inline base/base/LineReader.cpp:41]: why this mutex is now needed? Looks unrelated to adding periodical update, to something else?

File: src/Client/ClientBase.cpp [hunk 3]
```diff
@@ -1709,6 +1708,7 @@ void ClientBase::init(int argc, char ** argv)
         ("profile-events-delay-ms", po::value<UInt64>()->default_value(profile_events.delay_ms), "Delay between printing `ProfileEvents` packets (-1 - print only totals, 0 - print every single packet)")
 
         ("interactive", "Process queries-file or --query query and start interactive mode")
+        ("suggest-interval-ms", po::value<UInt64>()->default_value(suggest_interval_ms), "Reload suggestions interval")
```
Comment (kssenii) [inline src/Client/ClientBase.cpp:1711]: Also need to add here https://github.com/ClickHouse/ClickHouse/blob/master/src/Client/ClientBase.cpp#L1785?

Comment (azat) [reply src/Client/ClientBase.cpp]: Ok, let's add an ability to specify this delay in client xml config. Done.

File: base/base/LineReader.cpp (hunk 2 above)
Comment (azat) [reply base/base/LineReader.cpp:41]: Completion is reloaded in a separate thread, and so to avoid race between read of `words_no_case`/`words` there is a mutex now here and here - https://github.com/ClickHouse/ClickHouse/pull/32370/files#diff-11fe0247064a69cca4819453e93874d9d73dff4feb242daee35dcf7e54f48f7aR168-R172

**Review by kssenii — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
Let's close it, because the benefits don't outweight the potential drawbacks.
